### PR TITLE
Add ArrayValue type

### PR DESCRIPTION
**Pull request by michalstruck**
Hey,
first time contributing, a simple type based on #271

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -76,19 +76,19 @@ PR welcome for additional commonly needed types and docs improvements. Read the
 npm install type-fest
 ```
 
-*Requires TypeScript >=5.1*
+_Requires TypeScript >=5.1_
```
Comment (sindresorhus) [inline readme.md:79]: Don't make unrelated changes.

**Comment by sindresorhus** [comment]
Make sure you adhere to: https://github.com/sindresorhus/type-fest/blob/main/.github/contributing.md#submitting-a-new-type

File: readme.md (hunk 1 above)
Comment (michalstruck) [reply readme.md]: My bad, forgot to turn Prettier off and didn't notice all the changes it made. All fixed now

**Comment by michalstruck** [comment]
> Make sure you adhere to: https://github.com/sindresorhus/type-fest/blob/main/.github/contributing.md#submitting-a-new-type

Is there anything else I forgot to add?

File: readme.md (hunk 1 above)
Comment (sindresorhus) [reply readme.md:79]: It's not fixed.

**Comment by sindresorhus** [comment]
> Is there anything else I forgot to add?

Simply go through the list and check yourself ;)

File: readme.md (hunk 1 above)
Comment (michalstruck) [reply readme.md]: Well, this was awkward. Now I don't see much in the readme diff. Sorry about that.

**Comment by michalstruck** [comment]
So, apart from my double-mistake in readme formatting, I don't think there's much else to add/fix, that's why I asked. I added a bit more docs,  'explained' how it works.

File: source/array-value.d.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * @example
+ * ```ts
+ * import type {IterableElement} from 'type-fest';
+ *
+ * type Bar = {dataPoint: number, timestamp: number}[];
+ * type ValueBar = ArrayValue<Bar>; // => {dataPoint: number, timestamp: number}
+ *```
+ *
+ */
+export type ArrayValue<TArray extends unknown[]> = IterableElement<TArray>;
```
Comment (sindresorhus) [inline source/array-value.d.ts:35]:
```suggestion
export type ArrayValue<Element extends unknown[]> = IterableElement<Element>;
```

File: source/array-value.d.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * It is advised that one uses other, built in language methods where applicable, e.g:
+ * ```ts
+ * const foo: string[] = [];
+ * type Value = typeof foo[number]; // => string
+ * ```
+ *
+ * The creation of this type was discussed here - {@link https://github.com/sindresorhus/type-fest/issues/271}
+ *
+ * @category Array
```
Comment (sindresorhus) [inline source/array-value.d.ts:17]: Should be at the end.

File: source/array-value.d.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * This type is useful when you want to extract all the possible element types of an array.
+ *
+ * This type is a narrowed down use case of {@link IterableElement}.
+ *
+ * It is advised that one uses other, built in language methods where applicable, e.g:
+ * ```ts
+ * const foo: string[] = [];
+ * type Value = typeof foo[number]; // => string
+ * ```
+ *
```
Comment (sindresorhus) [inline source/array-value.d.ts:14]: Don't prefix the documentation comment with `*`. See other types as example.

File: source/array-value.d.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import type {IterableElement} from './iterable-element';
+/**
+ * Get a union of all the possible array/tuple values, with improved readability.
```
Comment (sindresorhus) [inline source/array-value.d.ts:3]: Unclear what "improved readability" means.

File: source/array-value.d.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ * type Value = typeof foo[number]; // => string
+ * ```
+ *
+ * The creation of this type was discussed here - {@link https://github.com/sindresorhus/type-fest/issues/271}
+ *
+ * @category Array
+ *
+ * @example
+ * ```ts
+ * import type {IterableElement} from 'type-fest';
```
Comment (sindresorhus) [inline source/array-value.d.ts:21]: Incorrect import.

File: source/array-value.d.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import type {IterableElement} from './iterable-element';
+/**
+ * Get a union of all the possible array/tuple values, with improved readability.
+ *
+ * This type is useful when you want to extract all the possible element types of an array.
+ *
+ * This type is a narrowed down use case of {@link IterableElement}.
+ *
+ * It is advised that one uses other, built in language methods where applicable, e.g:
```
Comment (sindresorhus) [inline source/array-value.d.ts:9]: This needs to be better written.

**Comment by sindresorhus** [comment]
Bump

**Comment by lucasteles** [comment]
@sindresorhus I believe this may be closed as #720 has been merged
